x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DeleteStep.java

```diff
@@ -41,7 +41,8 @@ public void performDuringNoSnapshot(IndexMetadata indexMetadata, ClusterState cu
 
         if (dataStream != null) {
             assert dataStream.getWriteIndex() != null : dataStream.getName() + " has no write index";
-            if (dataStream.getIndices().size() == 1 && dataStream.getIndices().get(0).equals(indexMetadata.getIndex())) {
+
+            if (dataStream.getIndices().size() == 1 && dataStream.getWriteIndex().getName().equals(indexName)) {
```

Comment:
Can you add a comment about why we use name equality here so it doesn't get accidentally changed back? (I know we have tests, but it's still easy to stop someone from wasting work)